ui/src/testing/react-test-helpers.ts

```diff
@@ -9,11 +9,18 @@ import * as fp from 'lodash/fp';
 // And fails with:
 //   ReferenceError: Zone is not defined
 export async function waitOneTickAndUpdate(wrapper: ReactWrapper) {
-  const waitImmediate = () => new Promise<void>(resolve => setImmediate(resolve));
+  // Make sure to use setTimeout. Combining setTimeout (used in UI) and setImmediate can result in a non-deterministic order of events
+  const waitImmediate = () => new Promise<void>(resolve => setImmediate(resolve))
   await act(waitImmediate);
   wrapper.update();
 }
 
+export async function waitOnTimersAndUpdate(wrapper: ReactWrapper){
```

Comment:
please add a comment to explain when this function should be used vs the other. I think based on what you're saying, the other should only be used with `fakeAsync`, and this should be used in all other cases.

Reply:
added comment